### PR TITLE
Sampling allocation bytes precisely - step1

### PR DESCRIPTION
**Pull request by LinHu2016**
Introduce APIs and fields needed by OMR's for allocation sampling.

related: https://github.com/eclipse/openj9/issues/7740

Signed-off-by: Lin Hu <linhu@ca.ibm.com>

### CODE REVIEW

**Comment by LinHu2016** [comment]
@amicic @dmitripivkine please review the change, Thanks

File: runtime/gc_glue_java/EnvironmentDelegate.hpp [hunk 1]
```diff
@@ -178,6 +178,28 @@ class MM_EnvironmentDelegate
...
+	/**
+	 * Restore heapTop from realHeapTop if realHeapTop != NULL
+	 */
+	void resetTLHSamplingTop() {}
+
+	/**
+	 * Retrieve allocation size inside TLH Cache.
+	 * @return (heapAlloc - heapBase)
+	 */
+	uintptr_t getAllocSizeInsideTLH() { return 0; }
```
Comment (amicic) [inline runtime/gc_glue_java/EnvironmentDelegate.hpp:201]: let's use passive: Allocated or Used

**Comment by amicic** [comment]
Start your initial comment with something like this (to be more desriptive). BTW, is there a high level Issue that covers this?

"Introduce APIs and fields needed by OMR's for allocation sampling."

**Comment by amicic** [comment]
Jenkins test compile win jdk11

**Comment by amicic** [comment]
Shorten the title/subject

**Comment by amicic** [comment]
Don't say: "Fixing..." - it may close the Issue.

Your commit message still has the long title (not sure if it's a problem, but let's just make it consistent with PR title).

**Pull request merged**